Dockerfile.spdk

```diff
@@ -0,0 +1,107 @@
+# syntax = docker/dockerfile:1.4
+ARG OS_RELEASE=9
+FROM registry.access.redhat.com/ubi$OS_RELEASE/ubi:latest as prebase
+ARG OS_RELEASE
+
+# Speed up dnf
+COPY <<EOF /etc/dnf/dnf.conf
+[main]
+gpgcheck=1
+installonly_limit=3
+clean_requirements_on_remove=True
+skip_if_unavailable=False
+best=False
+fastestmirror=True
+max_parallel_downloads=10
+tsflags=nodocs
+EOF
+
+#------------------------------------------------------------------------------
+FROM prebase as base_release_8
+ARG CENTOS_PACKAGE_VERSION="8-6"
+ARG CENTOS_MIRROR="mirror.centos.org/centos"
+
+#------------------------------------------------------------------------------
+FROM prebase as base_release_9
+ARG CENTOS_PACKAGE_VERSION="9.0-20"
+ARG CENTOS_MIRROR="mirror.stream.centos.org"
+
+#------------------------------------------------------------------------------
+FROM base_release_$OS_RELEASE as base
+
+ARG CENTOS_URL="http://$CENTOS_MIRROR/$OS_RELEASE-stream/BaseOS/x86_64/os/Packages"
+ARG CENTOS_GPG_PACKAGE="$CENTOS_URL/centos-gpg-keys-$CENTOS_PACKAGE_VERSION.el$OS_RELEASE.noarch.rpm"
+ARG CENTOS_REPO_PACKAGE="$CENTOS_URL/centos-stream-repos-$CENTOS_PACKAGE_VERSION.el$OS_RELEASE.noarch.rpm"
```

Comment:
This is complex to add `centos-stream-release` on to the ubi9 images. Some things I found that will lead to further maintenance:
- this uses plaintext http instead of https
- it uses pinned version numbers (`9.0-20` when `9.0-21` is out)
- `x86_64` is hard-coded into `CENTOS_URL`.  It's technically fine on non-x86 arches since the `centos-stream-repos` package is `noarch`, but that's a bit confusing on first reading

Maybe there's another way to approach this. What packages do you need from CentOS repos that are not available in the built-in ubi9 Yum repository?

I should probably try building this so I get a feel for it.

Reply:
> This is complex to add `centos-stream-release` on to the ubi9 images. Some things I found that will lead to further maintenance:

Yeah, I don't like this too much either, but other approaches were even less maintainable (like directly installing remote RPMs)... I also tried using RockyLinux, since CentOS Stream is not the natural fit for UBI, but discarded that for different reasons.

> * this uses plaintext http instead of https

Unfortunately, CentOS mirrors [don't provide HTTPS support](https://serverfault.com/questions/1089416/why-are-centos-mirrors-http-and-not-https).

> * it uses pinned version numbers (`9.0-20` when `9.0-21` is out)

There's no way to solve this. Being a rolling release, sooner or later package versions + builds will change...

> * `x86_64` is hard-coded into `CENTOS_URL`.  It's technically fine on non-x86 arches since the `centos-stream-repos` package is `noarch`, but that's a bit confusing on first reading

Yeah, I was also surprised by this, but they don't provide a `noarch` dir :/

> Maybe there's another way to approach this. What packages do you need from CentOS repos that are not available in the built-in ubi9 Yum repository?
> 
> I should probably try building this so I get a feel for it.

Personally, for the build stage I'd use CentOS container as base image, since I'm not using the building stage for anything other than generating RPMs that are later copied on top of the final UBI base layer. I'll check what would be the missing packages in the final container.